extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/context/ArcContextProvider.java

```diff
@@ -141,11 +141,21 @@ public ThreadContextController begin() {
             if (toRestore != null) {
                 // context active, store current state, feed it new one and restore state afterwards
                 // it is not necessary to deactivate the context first - just overwrite the previous state
-                requestContext.activate(state);
+                if (state.isValid()) {
```

Comment:
I like the less verbose version you've used below more ;-)... `requestContext.activate(state.isValid() ? state : null)`

Reply:
Good point, I meant to change that but forgot. Will do!